saas_public/tests/test_saas_public.py

```diff
@@ -1,15 +1,25 @@
-# Copyright 2018-2019 Denis Mudarisov <https://it-projects.info/team/trojikman>
+# Copyright 2019 Denis Mudarisov <https://it-projects.info/team/trojikman>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo.addons.saas.tests.common_saas_test import Common, DB_TEMPLATE_1, DB_TEMPLATE_2
 from odoo.tests.common import tagged, HttpCase
 from odoo.service import db
 
 
-# @tagged('post_install', 'at_install')
-# class TestSaas(HttpCase):
-#     def test_saas_public(self):
-#         template_id = self.saas_template_1.id
-#         url = '/saas/{}/create-fast-build'.format(template_id)
-#         r = self.url_open(url)
-#         self.assertEqual(r.status_code, 200, 'User must be redirected to the build')
-#         self.assertIn('fast_build_001', db.list_dbs())
-#         self.assert_no_error_in_db('fast_build_001')
+@tagged('post_install', 'at_install')
+class TestSaasPublic(HttpCase, Common):
+    def setUp(self):
+        super(TestSaasPublic, self).setUp()
+        self.setup_saas_env()
+
+    def test_saas_public(self):
+        if DB_TEMPLATE_1 in db.list_dbs():
+            db.exp_drop(DB_TEMPLATE_1)
+        if DB_TEMPLATE_2 in db.list_dbs():
+            db.exp_drop(DB_TEMPLATE_2)
+        if 'template_database' in db.list_dbs():
+            db.exp_drop('template_database')
+        self.env['saas.template.operator'].preparing_template_next()
+        template_id = self.saas_template_1.id
+        url = '/saas_public/{}/create-fast-build'.format(template_id)
+        r = self.url_open(url)
+        self.assertIn(r.status_code, [200, 204], 'User must be redirected to the build')
```

Comment:
Add another test to check ``public_access`` setting (it should return 404\403)

Reply:
In fact, I now realized that this is an incorrect test, because `saas_template_1` doesn't have `public_access =  True`